File: engine/src/main/java/org/epdb/engine/dto/Tuple.java
package org.epdb.engine.dto;

import java.util.Arrays;

public record Tuple(ColumnValue[] values) {

    public ColumnValue getValueAtIndex(int index) {
        return this.values()[index];
    }

    @Override
    public String toString() {
        return Arrays.toString(values);
    }
}

File: engine/src/main/kotlin/org/epdb/engine/dto/Tuple.kt
package org.epdb.engine.dto

import org.epdb.engine.columntypes.ColumnValue

data class Tuple(val values: List<ColumnValue>) {

    fun getValueAtIndex(index: Int): ColumnValue {
        return this.values[index]
    }

    override fun toString(): String {

        return values.toString()
    }
}
